src/language_concepts/data_types.md

```diff
@@ -1,206 +1,192 @@
-    let res = x * N;
+An integer type is a range constrained `Field` value, and the Noir frontend currently supports unsigned, arbitrary-sized integer types.
 
-    constrain res == y[0];
+An integer type is specified first with the letter `u`, indicating its unsigned nature, followed by its length in bits (e.g. `32`). For example, a `u32` variable can store a value in the range of \\([0,2^{32}-1]\\):
 
-    let res2 = x * mysubmodule::N;
-    constrain res != res2;
+```rust,noplaypen
+fn main(x : Field, y : u32) {
+    let z = x as u32 + y;
 }
+```
 
-mod mysubmodule {
-    use dep::std;
-
-    global N: Field = 10;
+`x`, `y` and `z` are all private values in this example. However, `x` is a `Field` while `y` and `z` are unsigned 32-bit integers. If `y` or `z` exceeds the range \\([0,2^{32}-1]\\), proofs created will be rejected by the verifier.
 
-    fn my_helper() -> comptime Field {
-        let x = N;
-        x
-    }
-}
-```
+> **Note:** The default TurboPlonk backend supports both even (e.g. `u16`, `u48`) and odd (e.g. `u5`, `u3`) sized integer types.
 
-#### `pub` Modifier
+### The Boolean Type
 
-A public type is a value that may change per circuit instance. Unlike Constants, changing a public value will not change the circuit.
+A boolean type in Noir has two possible values: `true` and `false`, and is specified using `bool`. For example:
 
 ```rust,noplaypen
-fn main(x : pub Field) {
-
+fn main() {
+    let t = true; // type implicitly interpreted
+    let f: bool = false; // type explicit annotated
 }
 ```
 
-As of the beta release, public types can only be declared through parameters on `main`.
-In the code, they are treated no differently to witness types.
+> **Note:** When returning a boolean value, it will show up as a value of 1 for `true` and 0 for `false` in _Verifier.toml_.
 
-> **Note:** This behaviour and type will change in future releases, to catch a linearity bug in user code.
+The main usage of the boolean type is in conditionals (e.g. `if` expressions, `constrain`). More about conditionals are covered in the [Control Flow](./control_flow.md) and [Constrain Statement](./constrain.md) sections.
 
 ## Compound Types
 
-Noir also supports arrays, structs, and tuples whose elements may be public or private.
+A compound type groups together multiple values into one type. Elements within a compound type can be private or public.
 
-#### Arrays
+### The Array Type
 
-An array is a data structure which allows you to group together data types.
-All values in an array must be of the same type (i.e., they're homogeneous).
+An array is one way of grouping together values into one compound type:
 
 ```rust,noplaypen
 fn main(x : Field, y : Field) {
     let my_arr = [x, y];
 }
 ```
 
-Note that currently Noir only supports arrays with integer or `Field` elements.
+All elements in an array must be of the same type (i.e. homogeneous). That is, `Field` values cannot be grouped together with integer values in one array for example.
```

Comment:
Since this is targetted towards beginners, lets make the example a bit less generic and use "... together with `u8` values" rather than "integer values" since that may lead to more questions for users confused with the difference between Fields and integers.